### PR TITLE
Mizar-Arktos integration with multitenancy network isolation for n-TP x n-RP scale out architecture

### PR DESCRIPTION
**Pull request by vinaykul**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
/kind feature
> /kind flake

**What this PR does / why we need it**:  This deploys scale-out cluster with multi-tenancy network isolation via Mizar for multiple tenant partitions and multiple resource partitions. It configures mizar operator to in arktos mode. It deploys mizar operators and daemons as static pods, and deploys the per-TP operator on the TP master node which achieves better performance while reducing complexity.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
Steps to deploy:
```
unset KUBE_GCE_MASTER_PROJECT KUBE_GCE_NODE_PROJECT KUBE_GCI_VERSION  KUBE_GCE_MASTER_IMAGE  KUBE_GCE_NODE_IMAGE KUBE_CONTAINER_RUNTIME NETWORK_PROVIDER DISABLE_NETWORK_SERVICE_SUPPORT

export SCALEOUT_CLUSTER=true NUM_NODES=2 SCALEOUT_TP_COUNT=1 SCALEOUT_RP_COUNT=1 RUN_PREFIX=vinay-mark

export MASTER_DISK_SIZE=500GB MASTER_ROOT_DISK_SIZE=500GB KUBE_GCE_ZONE=us-west2-b MASTER_SIZE=n1-highmem-32 NODE_SIZE=n1-highmem-16 NODE_DISK_SIZE=500GB GOPATH=$HOME/go KUBE_GCE_ENABLE_IP_ALIASES=true KUBE_GCE_PRIVATE_CLUSTER=true CREATE_CUSTOM_NETWORK=true KUBE_GCE_INSTANCE_PREFIX=${RUN_PREFIX} KUBE_GCE_NETWORK=${RUN_PREFIX} ENABLE_KCM_LEADER_ELECT=false ENABLE_SCHEDULER_LEADER_ELECT=false ETCD_QUOTA_BACKEND_BYTES=8589934592 SHARE_PARTITIONSERVER=false LOGROTATE_FILES_MAX_COUNT=200 LOGROTATE_MAX_SIZE=200M KUBE_ENABLE_APISERVER_INSECURE_PORT=true KUBE_ENABLE_PROMETHEUS_DEBUG=true KUBE_ENABLE_PPROF_DEBUG=true TEST_CLUSTER_LOG_LEVEL=--v=2 HOLLOW_KUBELET_TEST_LOG_LEVEL=--v=2 GCE_REGION=us-west2-b

NETWORK_PROVIDER=mizar ./cluster/kube-up.sh
```

I also tried this for 2-TP 2-RP case with pods in Mizar default VPC and subnet (vpc0/net0) and I can see Mizar providing pod network connectivity with multi-tenancy network isolation for overlapping IPs across two different tenant partitions with pods scheduled across different RPs. 

However, system-default-vpc/subnet is not automatically created, and any tenant created via API does not trigger creation of tenant-specific network, VPC, subnet objects.

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cluster/gce/gci/configure-helper-common.sh [hunk 1]
```diff
@@ -3214,6 +3214,459 @@ function start-cluster-networking {
...
+          jsonPath: .spec.createtime
+          description: Time the object is created
+        - name: ProvisionDelay
+          type: string
+          priority: 0
+          jsonPath: .spec.provisiondelay
+          description: Time to provision an object from creation
+EOF
+}
+
```
Comment (sonyafenge) [inline cluster/gce/gci/configure-helper-common.sh:3669]: suggest to save this yaml file in code and move to master machines as other static pods yaml/manifest. or download this yaml from git repo.

File: cluster/gce/gci/configure-helper-common.sh [hunk 2]
```diff
@@ -2353,7 +2353,7 @@ function start-kube-controller-manager {
     RUN_CONTROLLERS="serviceaccount,serviceaccount-token,nodelifecycle,nodeipam,ttl,csrsigning,csrapproving,csrcleaner"
   fi
   if [[ "${ARKTOS_SCALEOUT_SERVER_TYPE:-}" == "tp" ]]; then
-    RUN_CONTROLLERS="*,-nodeipam,-nodelifecycle,-mizar-controllers,-network,-ttl"
```
Comment (h-w-chen) [inline cluster/gce/gci/configure-helper-common.sh:2356]: is there specific reason to remove "-mizar-controllers, -network" here?

File: cluster/gce/gci/configure-helper-common.sh [hunk 3]
```diff
@@ -3214,6 +3214,459 @@ function start-cluster-networking {
   esac
 }
 
+function create-mizar-crds-yaml {
```
Comment (Sindica) [inline cluster/gce/gci/configure-helper-common.sh:3217]: Please use a separate yaml file instead

File: cluster/gce/gci/configure-helper-common.sh [hunk 4]
```diff
@@ -3214,6 +3214,459 @@ function start-cluster-networking {
...
+        - name: ProvisionDelay
+          type: string
+          priority: 0
+          jsonPath: .spec.provisiondelay
+          description: Time to provision an object from creation
+EOF
+}
+
 function create-mizar-daemon-manifest {
   cat <<EOF >/etc/kubernetes/manifests/mizar-daemon.yaml
```
Comment (Sindica) [inline cluster/gce/gci/configure-helper-common.sh:3671]: Same here

**Review by h-w-chen — APPROVED** [review]
approved on assumption all the feed backs would be addressed.

File: cluster/gce/gci/configure-helper-common.sh (hunk 1 above)
Comment (vinaykul) [reply cluster/gce/gci/configure-helper-common.sh:3669]: I tried to do this following the network template example but it does not seem to be working. I might be missing something. Perhaps we can fix this after 1/30 release.

File: cluster/gce/gci/configure-helper-common.sh (hunk 2 above)
Comment (vinaykul) [reply cluster/gce/gci/configure-helper-common.sh:2356]: Mizar controller needs to run in the TP doesn't it?

File: cluster/gce/gci/configure-helper-common.sh (hunk 4 above)
Comment (vinaykul) [reply cluster/gce/gci/configure-helper-common.sh:3671]: I tried this .. spent a day getting it to work but I'm missing something. Perhaps someone in arktos team can do this? I need to focus on CNI fixes.

File: cluster/gce/gci/configure-helper-common.sh (hunk 2 above)
Comment (Sindica) [reply cluster/gce/gci/configure-helper-common.sh:2356]: This is aligned with local dev config.

File: cluster/gce/gci/configure-helper-common.sh (hunk 4 above)
Comment (vinaykul) [reply cluster/gce/gci/configure-helper-common.sh]: Fixed with Sonya's help.

File: cluster/gce/gci/configure-helper-common.sh (hunk 3 above)
Comment (vinaykul) [reply cluster/gce/gci/configure-helper-common.sh]: Fixed with Sonya's help

File: cluster/gce/gci/configure-helper-common.sh (hunk 1 above)
Comment (vinaykul) [reply cluster/gce/gci/configure-helper-common.sh]: fixed with your change.

**Review by sonyafenge — APPROVED** [review]
/LGTM

Tested scale-out 1*1 and all mizar component started successfully

File: cluster/gce/config-test.sh [hunk 5]
```diff
@@ -95,8 +95,10 @@ fi
 NETWORK_PROVIDER="${NETWORK_PROVIDER:-cni}" # none, kubenet doesn't work with container runtime, update to cni and flannel
 
 if [[ "${NETWORK_PROVIDER:-}" == "mizar" ]]; then
-  NETWORK_PROVIDER_VERSION="0.9"
+  NETWORK_PROVIDER_VERSION="dev"
```
Comment (Sindica) [inline cluster/gce/config-test.sh:98]: This need to be changed when merging back to master

File: cluster/gce/gci/configure-helper-common.sh [hunk 6]
```diff
@@ -3301,21 +3233,53 @@ function wait-until-mizar-ready {
...
+  echo "Installing Mizar for scale-out architecture..."
+  local -r src_dir="${KUBE_HOME}/kube-manifests/kubernetes/gci-trusty"
+  mkdir -p "${KUBE_HOME}/mizar"
+
+  echo "Deploying mizar daemonset"
+  # Place mizar daemon yaml.
+  src_file="${src_dir}/mizar-daemon.yaml"
+  sed -i -e "s@{{network_provider_version}}@${NETWORK_PROVIDER_VERSION}@g" "${src_file}"
+  mv "${src_dir}/mizar-daemon.yaml" "${KUBE_HOME}/mizar/"
+  kubectl apply -f "${KUBE_HOME}/mizar/mizar-daemon.yaml"
```
Comment (Sindica) [inline cluster/gce/gci/configure-helper-common.sh:3262]: start-mizar-scaleout is used in all scaleout tp cluster. It deploys mizar daemon as well as crds and operators. The later two can be deployed to all tps but not daemon. Is there anything prevent mizar daemon from being deployed more than one tp?

File: cluster/gce/config-test.sh (hunk 5 above)
Comment (sonyafenge) [reply cluster/gce/config-test.sh:98]: @vinaykul Could you please explain this? I changed here to match config-test same as config-default.

File: cluster/gce/gci/configure-helper-common.sh (hunk 6 above)
Comment (sonyafenge) [reply cluster/gce/gci/configure-helper-common.sh:3262]: Daomon can be deployed only from tp1 if this is the agreement.

File: third_party/mizar/mizar-daemon.yaml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        # The daemon shall run on the master node
+        - effect: NoSchedule
+          operator: Exists
+      terminationGracePeriodSeconds: 5
+      serviceAccountName: mizar-operator
+      hostNetwork: true
+      hostPID: true
+      initContainers:
+      - name: node-init
+        image: mizarnet/mizar:{{network_provider_version}}
```
Comment (Sindica) [inline third_party/mizar/mizar-daemon.yaml:26]: network_provider_version is better named as mizar_version

**Pull request merged**